crates/orderbook/src/orderbook.rs

```diff
@@ -68,20 +67,42 @@ impl Metrics {
             .expect("unexpected error getting metrics instance")
     }
 
-    fn on_order_operation(order: &Order, operation: OrderOperation) {
-        let class = order_class_label(&order.metadata.class);
-        let op = operation_label(&operation);
-        Self::get().orders.with_label_values(&[class, op]).inc();
+    fn on_order_operation(order: &Order, operation: OrderOperation, quote: Option<&Quote>) {
+        let class = match quote {
```

Comment:
I find the `match` construct a bit weird to read.
Played a bit around and ended up with:
```rust
        let class = if quote.is_some_and(|quote| {
            !is_order_outside_market_price(
                &Amounts {
                    sell: order.data.sell_amount,
                    buy: order.data.buy_amount,
                    fee: order.data.fee_amount,
                },
                &Amounts {
                    sell: quote.sell_amount,
                    buy: quote.buy_amount,
                    fee: quote.fee_amount,
                },
                order.data.kind,
            )
        }) {
            OrderClass::User
        } else {
            OrderClass::Limit
        };
```

I don't like that much more so feel free to stick with your version if you prefer that.